File: algorithms/121.Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int buy = 0;
        int profit = 0;
        for (int i = 0; i < prices.size(); ++i) {
            if (prices[buy] > prices[i])
                buy = i;
            profit = max(profit, prices[i] - prices[buy]);
        }
    }
};

int main() {

    return 0;
}

File: algorithms/121.Best_Time_to_Buy_and_Sell_Stock/best-time-to-buy-and-sell-stock.go
package problem0121

func maxProfit(prices []int) int {

	var max int = 0
	var tmp int = 0

	for i := 1; i < len(prices); i++ {
		tmp += prices[i] - prices[i-1]
		if tmp < 0 {
			tmp = 0
		}
		if max < tmp {
			max = tmp
		}
	}

	return max
}
